### DISCUSSION TITLE
Passing token in the headers

### DISCUSSION THREAD

**Discussion by ZouTiger** [gladly-team/next-firebase-auth#55] [Q&A] [answered]
Great project first of all so thank you for creating it. 

I am working with Hasura, which requires JWT token to be included in the headers. How would I do that using next-firebase-auth? Is there a way to access the token on client so I can pass it to Apollo Client?

Hasura also requires custom claims but that I can solve by using Hasura's claims_map.

**Comment by tlays** [comment, marked as answer]
Hi @ZouTiger , I believe this is the example that you are looking for: https://github.com/gladly-team/next-firebase-auth/blob/main/example/pages/static-auth-required-loader.js

1. Make sure that page is wrapped with `withAuthUser` HOC
2. Then `const AuthUser = useAuthUser()` to retrieve the instance of the currently logged in user
3. `const token = await AuthUser.getIdToken()` to retrieve auth token
4. And then you can call your external API with the headers properly set as done [here](https://github.com/gladly-team/next-firebase-auth/blob/b581d97738ea03d8819d9f64f2a28ce474f73387/example/pages/static-auth-required-loader.js#L24)

**Comment by ZouTiger** [comment]
@tlays Many thanks! Apparently, I wasn't paying attention enough.
